File: Java/src/main/java/QuestionBank/Solution201_300/Solution226.java
package QuestionBank.Solution201_300;

import tool.TreeNode;

/**
 * @author windows
 */
public class Solution226 {
    public TreeNode invertTree(TreeNode root) {
        if (root == null){
            return root;
        }
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;
        invertTree(root.left);
        invertTree(root.right);
        return root;
    }


    public static void main(String[] args) {
        Solution226 solution226 = new Solution226();
    }
}

File: C++/src/Solution201_300/Solution226.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution226.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

TreeNode *Solution226::invertTree(TreeNode *root) {
    dfs(root);
    return root;
}

void Solution226::dfs(TreeNode *root){
    if (root == nullptr) return;
    TreeNode* temp = root->left;
    root->left = root->right;
    root->right = temp;
    dfs(root->left);
    dfs(root->right);
}
